include/eve/detail/function/simd/arm/sve/slice.hpp

```diff
@@ -0,0 +1,57 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/detail/category.hpp>
+#include <eve/detail/implementation.hpp>
+
+namespace eve::detail
+{
+//================================================================================================
+// Single slice
+//================================================================================================
+template<typename T, typename N, typename Slice>
+EVE_FORCEINLINE wide<T, typename N::split_type>
+                slice(wide<T, N> a, Slice) noexcept requires sve_abi<abi_t<T, N>>
+{
+  if constexpr( !Slice::value ) return a.storage();
+  else return svext(a, a, N::value / 2);
+}
+
+template<typename T, typename N, typename Slice>
+EVE_FORCEINLINE logical<wide<T, typename N::split_type>>
+                slice(logical<wide<T, N>> a, Slice) noexcept requires sve_abi<abi_t<T, N>>
+{
+  if constexpr( !Slice::value ) return a.storage();
+  else
+  {
+    auto bits = a.bitmask();
+    return svcmpne(svptrue_b8(), slice(bits, Slice{}), 0);
```

Comment:
Couldn't come up with anything. They gave us nothing to move bits around

Reply:
Can you clarify what is going on ?

Reply:
I convert to bits, slice and convert back?

Reply:
OK, I got distracted byt the raw svcmpne in the middle. Got it now